lib/logstorage/block_stream_merger.go

```diff
@@ -117,6 +118,14 @@ func (bsm *blockStreamMerger) mustInit(bsw *blockStreamWriter, bsrs []*blockStre
 	heap.Init(&bsm.readersHeap)
 }
 
+var mergeStreamsExceedLogger = logger.WithThrottler("mergeStreamsExceed", 10*time.Second)
+
+func (bsm *blockStreamMerger) mergeStreamsLimitWarn(bd *blockData) {
```

Comment:
I think, warn message isn't relevant for user. It's better to introduce limitation during data ingestion and doesn't accept a block with columns exceed limit.

Reply:
Such limit was added in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4976, but it is still possible to face this issue in case of incorrectly configured log shipper.
If log will be parsed in a way which will create a lot of unique fields with small size, but amount of fields will be less than `-insert.maxFieldsPerLine` for each line over time total amount of such fields will still trigger this problem.